src/io/matpower.jl

```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
+        println(io, "%% bus names")
+        println(io, "mpc.bus_name = {")
+        for (idx,bus) in sort(buses)
+            println(io, "\t'", bus["bus_name"], "'") 
+        end    
+        println(io, "};")
+        println(io)
+    end
+    
+    # Print the generator data
+    println(io, "%% generator data")
+    println(io, "%    bus    Pg    Qg    Qmax    Qmin    Vg    mBase    status    Pmax    Pmin    Pc1    Pc2    Qc1min    Qc1max    Qc2min    Qc2max    ramp_agc    ramp_10    ramp_30    ramp_q    apf")
+    println(io, "mpc.gen = [")
+    i = 1
+    for (idx,gen) in sort(generators)
+        if idx != gen["index"]
+            warn(LOGGER, "The index of the generator does not match the matpower assigned index. Any data that uses generator indexes for reference is corrupted.");           
+        end  
+        println(io, "\t", gen["gen_bus"], 
+                    "\t", (gen["pg"]*mvabase),
+                    "\t", (gen["qg"]*mvabase),
+                    "\t", (gen["qmax"]*mvabase),
+                    "\t", (gen["qmin"]*mvabase),
+                    "\t", gen["vg"],
+                    "\t", gen["mbase"],
+                    "\t", gen["gen_status"],
+                    "\t", (gen["pmax"]*mvabase),
+                    "\t", (gen["pmin"]*mvabase),
+                    "\t", gen["pc1"],
+                    "\t", gen["pc2"],
+                    "\t", gen["qc1min"],
+                    "\t", gen["qc1max"],
+                    "\t", gen["qc2min"],
+                    "\t", gen["qc2max"],
+                    "\t", gen["ramp_agc"],
+                    "\t", (haskey(gen, "ramp_10") ? gen["ramp_10"] : 0),
+                    "\t", gen["ramp_30"],
+                    "\t", gen["ramp_q"], 
+                    "\t", gen["apf"],
+                    "\t", (haskey(gen, "mu_pmax") ? gen["mu_pmax"] : ""),
+                    "\t", (haskey(gen, "mu_pmin") ? gen["mu_pmin"] : ""),
+                    "\t", (haskey(gen, "mu_qmax") ? gen["mu_qmax"] : ""),
+                    "\t", (haskey(gen, "mu_qmin") ? gen["mu_qmin"] : ""),
+                    )      
+        i = i+1
+    end
+    println(io,"];")
+    println(io)
+    
+    # Print the branch data
+    println(io, "%% branch data")
+    println(io, "%    fbus    tbus    r    x    b    rateA    rateB    rateC    ratio    angle    status    angmin    angmax")
+    println(io, "mpc.branch = [")
+    i = 1
+    for (idx,branch) in sort(branches)
+        if idx != branch["index"]
+            warn(LOGGER, "The index of the branch does not match the matpower assigned index. Any data that uses branch indexes for reference is corrupted.");           
+        end 
+        println(io, "\t", branch["f_bus"], 
+                    "\t", branch["t_bus"], 
+                    "\t", branch["br_r"], 
+                    "\t", branch["br_x"], 
+                    "\t", (haskey(branch,"b_to") ? branch["b_to"] + branch["b_fr"]  : 0), 
```

Comment:
I don't believe `b_to` is not optional, I think you can hard code this as `branch["b_to"] + branch["b_fr"]`